### DISCUSSION TITLE
PRs with unresolved comments are auto-merged despite protection rules

### DISCUSSION THREAD

**Discussion by rjhuijsman** [community/community#10076] [Pull Requests] [answered]
Our repositories have their branch protection rules set to "Require conversation resolution before merging".

We're noticing that when we enable auto-merge, this branch protection rule is ignored. [See this example](https://github.com/3rdparty/dev-tools/pull/21). In that example the interface even makes it look like the approval happens _after_ the auto-merge (race condition?).

We don't want a merge to happen as long as there are still unresolved comments - we treat "approve with comments" as a signal that "you should address this, but we don't need to re-review afterwards".

Having auto-merge violate the branch protection rule is preventing us from using auto-merge.

**Comment by srabraham** [comment]
(not an answer, but a clarification)
I've noticed this too, and it's definitely a blocker on our company using auto-merge. I'd add that this even happens when the unresolved conversation was made long before the approval, so the problem can't just be a race condition between conversation being posted and auto-merge being triggered.

**Comment by rjhuijsman** [comment]
For others dealing with this issue, we are happy using the third-party mergequeue.com product to solve this for us, avoiding the need for auto-merge. Its headliner merge queue functionality is very nice to have too. :)

**Comment by srabraham** [comment, marked as answer]
This now appears to be fixed in production.

I filed a bug report against GitHub for this issue last week, and they said that as of yesterday this is resolved. I verified that successfully for myself this morning.

**Reply by rjhuijsman** [reply]
Thanks @srabraham - that's great news!

For our education, where/how did you file the bug report against GitHub? Is that a thing that your company has premium access to, or would we have been able to do the same?

**Reply by srabraham** [reply]
Hi @rjhuijsman,
I filed the report through here: https://support.github.com/contact/bug-report
I did specify my corporate identity, rather than "my personal account", in the "Account or organization" field. I figured that could only help the report's priority, though I don't ultimately know what impact it had.

**Reply by rjhuijsman** [reply]
Super helpful, thank you! 🎉
